docs/src/pages/components/data-grid/editing/ValidateServerNameGrid.js

```diff
@@ -47,20 +47,26 @@ export default function ValidateServerNameGrid() {
         clearTimeout(promiseTimeout);
         clearTimeout(keyStrokeTimeoutRef.current);
 
-        apiRef.current.setEditCellProps({
-          id,
-          field,
-          props: { ...props, error: true },
+        let oldModel = apiRef.current.getEditRowsModel();
+        apiRef.current.setEditRowsModel({
```

Comment:
Would it make sense to introduce a callback version of `setEditRowsModel` ?

```suggestion
        apiRef.current.setEditRowsModel(oldModel => ({
          ...oldModel,
          [id]: {
            ...oldModel[id],
            [field]: { ...oldModel[id][field], error: true },
          },
        });
```

It would reduce the risk of a user passing an outdated model.

Reply:
We could, but in the docs we encourage the use of the `editRowsModel` prop, which doesn't suffer from this outdated-model risk. The imperative API is kinda hidden. We can keep the callback version in the backlog because there're others `setSomethingModel` methods and, for consistency, they should also support it.